src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -6473,26 +6474,7 @@ private static List<CompletionResult> GetSpecialHashTableKeyMembers(params strin
 
         internal static PowerShell AddCommandWithPreferenceSetting(PowerShell powershell, string command, Type type = null)
         {
-            Diagnostics.Assert(powershell != null, "the passed-in powershell cannot be null");
-            Diagnostics.Assert(!String.IsNullOrWhiteSpace(command), "the passed-in command name should not be null or whitespaces");
-
-            if (type != null)
-            {
-                var cmdletInfo = new CmdletInfo(command, type);
-                powershell.AddCommand(cmdletInfo);
-            }
-            else
-            {
-                powershell.AddCommand(command);
-            }
-            powershell
-                .AddParameter("ErrorAction", ActionPreference.Ignore)
-                .AddParameter("WarningAction", ActionPreference.Ignore)
-                .AddParameter("InformationAction", ActionPreference.Ignore)
-                .AddParameter("Verbose", false)
-                .AddParameter("Debug", false);
-
-            return powershell;
+            return TypeInferenceContext.AddCommandWithPreferenceSetting(powershell, command, type);
```

Comment:
Moving this method to `TypeInferenceContext` feels wrong - this is a generic utility method for completion, it's used for more than type inference.